### PR TITLE
Converts to TS

### PR DESCRIPTION
**Pull request by mex**
[sc-90328]
[skip-sc]

### CODE REVIEW

**Review by MisanthropicBit — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -4,13 +4,25 @@
...
+    "node": ">=16.0.0"
   },
   "scripts": {
-    "test": "nyc --reporter=lcov --reporter=text mocha \"src/**/*.test.js\"",
-    "coveralls": "nyc report --reporter=text-lcov | coveralls"
+    "test": "TZ=UTC jest",
+    "cov": "TZ=UTC jest --coverage=true",
+    "build": "build src",
+    "build:js": "build --skip-typings src",
+    "build:types": "tsc --noEmit",
```
[the remark this answers is not in the document]
Comment (mex) [reply package.json:14]: This is the same commands as other TS projects (generated from node-setup).

[the remark this answers is not in the document]
Comment (MisanthropicBit) [reply package.json:14]: I can find a few exceptions to this amongst our TS projects that call `--emitDeclarationOnly` and still use `build src` or similar as their build command. I'm not really too concerned about this just wanted to make sure it was correct and declarations are still emitted.

I guess all your projects will be aligned by [this story](https://app.shortcut.com/connectedcars/story/22950/all-pods-should-start-node-on-pid-1) anyway.

**Review by MisanthropicBit — APPROVED** [review]

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (MisanthropicBit) [reply package.json:14]: Actually looking into this a bit more, the `build:js` and `build:types` aren't used by the `build src` command from node-setup. Removing the build directory and running `build:types` gives me only a `tsconfig.tsbuildinfo` file in `build/dist/` whereas `--emitDeclarationOnly` gives me that file and all the declaration files in `build/dist/src/` so I think passing `--noEmit` here is actually not what this command should do.

On a related note, I'm not sure what the purpose of these extra build scripts are in the first place or if they are legacy.

[the remark this answers is not in the document]
Comment (MisanthropicBit) [reply package.json]: Great. I'll update node-vinutils accordinly.

**Review by MisanthropicBit — APPROVED** [review]

**Pull request merged**